compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1224,7 +1224,10 @@ class TreeUnpickler(reader: TastyReader,
               val alias = if currentAddr == end then EmptyTree else readTpt()
               TypeBoundsTree(lo, hi, alias)
             case HOLE =>
-              readHole(end, isType = false)
+              val idx = readNat()
+              val tpe = readType()
+              val args = until(end)(readTerm())
+              TreePickler.Hole(true, idx, args).withType(tpe)
```

Comment:
It seems now we can remove the `splices` param for `TreeUnpickler`.

Reply:
Yes :)

Reply:
Actually the `QuoteUnpickler` is redundant now. We can use the `DottyUnpickler` directly.